ultradns-rest/src/main/java/denominator/ultradns/util/RegionUtil.java

```diff
@@ -0,0 +1,98 @@
+package denominator.ultradns.util;
+
+import denominator.ultradns.model.Region;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.TreeSet;
+
+public final class RegionUtil {
+
+    // This class is not supposed to be instantiated.
+    private RegionUtil() { }
+
+    public static Map<String, Collection<String>> getRegionHierarchy(Region region) {
```

Comment:
The logic inside this method & getRegionHierarchyAsRegions seems to be very similar. Please refactor the code in this class to avoid any code duplication.